plugin/k8s_external/external.go

```diff
@@ -68,8 +73,11 @@ func (e *External) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Ms
 		return plugin.NextOrFailure(e.Name(), e.Next, ctx, w, r)
 	}
 
-	if e.externalFunc == nil {
-		return plugin.NextOrFailure(e.Name(), e.Next, ctx, w, r)
+	// TODO: If Kubernetes is not loaded, return an error directly on startup.
+	if !e.kubernetesPluginLoaded {
+		log.Error("the kubernetes plugin failed to load or misconfigured")
```

Comment:
We should not log an error here explicitly with a logger.  We should only return the servfail.